language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -336,9 +336,9 @@ tyReduce ty@(TyVar _ (TyNameWithKind (TyName (Name _ _ u)))) = do
         -- we must use recursive lookups because we can have an assignment
         -- a -> b and an assignment b -> c which is locally valid but in
         -- a smaller scope than a -> b.
-        Just ty'@TyVar{} -> tyReduce ty'
-        Just ty'         -> NormalizedType <$> cloneType ty'
-        Nothing          -> pure $ NormalizedType ty
+        Just ty'@(NormalizedType TyVar{}) -> pure ty'
```

Comment:
That seems like a very strange change. There is no loop that is justified in the comment right above.

Reply:
Type reduction should be idempotent if we have normalized types, no?

Reply:
Oh, I see what you mean. This is indeed bad.

Reply:
Your comment says

```
        -- we must use recursive lookups because we can have an assignment
        -- a -> b and an assignment b -> c which is locally valid but in
        -- a smaller scope than a -> b.
```

Where are those recursive lookups now?

Reply:
I don't think they are necessary, but I will have to find a test case or justification. 

I suspect the reason this failed the test suite in the past was due to not reducing within `(fix a _)` and thus was not idempotent like it was supposed to be.